apps/web/src/pages/integrations/IntegrationsStorePage.tsx

```diff
@@ -218,7 +218,9 @@ function initializeProviders(integrations: IntegrationEntity[]): IIntegratedProv
     if (integration?.credentials && Object.keys(clonedCredentials).length !== 0) {
       clonedCredentials.forEach((credential) => {
         // eslint-disable-next-line no-param-reassign
-        credential.value = integration.credentials[credential.key]?.toString();
+        if (credential.type === 'object') {
+          credential.value = JSON.stringify(integration.credentials[credential.key]);
+        } else credential.value = integration.credentials[credential.key]?.toString();
```

Comment:
```suggestion
        if (credentialcredential.type === 'object') {
          credential.value = JSON.stringify(integration.credentials[credential.key]);
        } else {
        credential.value = integration.credentials[credential.key]?.toString();
       }
```
Just to clarify the else.